bpf/lib/encap.h

```diff
@@ -8,6 +8,9 @@
 #include "dbg.h"
 #include "trace.h"
 #include "l3.h"
+#ifdef STRICT_MODE
```

Comment:
Calling this 'strict mode' creates a few smells:
- it claims a rather generic term to denote a WireGuard-specific behaviour
- it's not explicitly described or documented anywhere what `STRICT_MODE` does or why it exists

Even in the README, it says `Enable WireGuard Pod2Pod strict mode`, which is really vague. I would at least call this something more specific, like  `WIREGUARD_STRICT_MODE` so the code is more self-descriptive.

Also (this should probably be discussed at a community meeting or a platform wider than this PR), it looks to me like Cilium does something insecure-by-default here (sending unencrypted packets with encryption enabled), and this should probably not be a flag at all. It should be the default, not a 'mode' the agent can be configured in.

Reply:
> Also (this should probably be discussed at a community meeting or a platform wider than this PR), it looks to me like Cilium does something insecure-by-default here (sending unencrypted packets with encryption enabled), and this should probably not be a flag at all. It should be the default, not a 'mode' the agent can be configured in.

For context, plese check the following two resources:
https://docs.cilium.io/en/v1.12/gettingstarted/encryption/#egress-traffic-to-not-yet-discovered-remote-endpoints-may-be-unencrypted

https://github.com/cilium/cilium.io/pull/179

I don't think it makes sense to enable this by default. Notably, the feature doesn't work on ENI/Azure, since on cloud providers, there there is no way to knowing what IPs (or IP range) might have to be encrypted in the future. In such a case, you either need to block all traffic to unknown IPs (which is what the current docs recommend, but is at odds with the allow-all by default network policy we have in Cilium), or live with potential traffic leak. 

There is a trade-off. Some users will want to always enable strict mode and/or have a network policy deny-everything-by-default, but my impression is that most of our users are fine with opportunistic encryption.

Reply:
> I would at least call this something more specific, like WIREGUARD_STRICT_MODE so the code is more self-descriptive.

Maybe `ENCRYPTION_STRICT_MODE` is better.

The issue applies to both WireGuard and IPSec btw, though this PR only introduces strict mode to WireGuard for now. Fundamentally, I could also be added to IPSec too, I believe.

Reply:
@gandro Thanks for the explanation, all makes sense. `ENCRYPTION_STRICT_MODE` sounds good. :+1:

Reply:
The `#ifdef` is renamed. Therefore, I'll resolve this conversation. If there are still architectural questions/comments, please re-open.